tools/c7n_azure/c7n_azure/query.py

```diff
@@ -105,15 +109,21 @@ def source_type(self):
 
     def resources(self, query=None):
         key = self.get_cache_key(query)
-        resources = self.augment(self.source.get_resources(query))
+        resources = self.source.get_resources(query)
         self._cache.save(key, resources)
         return self.filter_resources(resources)
 
-    def get_resources(self, resource_ids):
-        resource_client = self.get_client('azure.mgmt.resource.ResourceManagementClient')
-        session = local_session(self.session_factory)
+    def get_resources(self, resource_ids, **params):
+        resource_client = self.get_client()
+        m = self.resource_type
+        get_client, get_op, extra_args = m.get_spec
```

Comment:
This is what I do and don't like about this implementation. In a way it's consistent with how we do the enumeration, but it's inconsistent with how they've implemented this in c7n for AWS.

Reply:
The old way this was implemented was just meant to be sort of a generic min-bar, I'm not sure it was used anywhere or would have been useful for anything - everyone had to reimplement/override it themselves.

Reply:
oic what you did now, this seems excellent... 

You always had the option of just overriding this method in any resource that needed a custom one (and still do) - but if this generic solution works for most resources then I like it.

Reply:
Yeah the previous implementation assumed all resources hitting this was an ARM resource, but other methods in this file didn't make that assumption. So there was a bit of a disconnect for me trying to figure out what was going on in this file. I decided to try and implement this the same way we implement enumerating since it seemed like it works for any resource.